python/lsst/daf/butler/registry/queries/_query.py

```diff
@@ -339,6 +562,33 @@ def explain_no_results(
                 yield from followup_query.explain_no_results(db, region=region, followup=False)
                 return
 
+    def order_by(self, db: Database, *args: str) -> None:
+        """Make query return ordered result.
+
+        Parameters
+        ----------
+        db : `Database`
+            Object managing the database connection.
+        args : `str`
+            Names of the columns/dimensions to use for ordering. Column name
+            can be prefixed with minus sign to use descending ordering.
+        """
+        self._order_by = _OrderByHandler(args, self.graph, db)
+
+    @abstractmethod
+    def limit(self, limit: int, offset: Optional[int] = None) -> None:
+        """Make query return limited number of records.
+
+        Parameters
+        ----------
+        limit : `int`
+            Upper limit on the number of returned records.
+        offset : `int` or `None`
+            If not `None` then the number of records to skip before returning
+            ``limit`` records.
+        """
+        raise NotImplementedError()
```

Comment:
Like `order_by`, this should return a new `Query` and not modify self (or at least `self` after modifying it).

Reply:
Is there a simpler way to clone Query than doing `makeBuilder`? It feels a bit heavy-weight for changing one simple attribute.